### ISSUE TITLE
[BUG] useScroll fails to keep track of Element scroll when the ref is passed down to a child component. ONLY happens in Production.

### ISSUE THREAD

**Issue by eeshankeni** [framer/motion#2483]
UseScroll has a Container prop which is used to keep track of the scroll of a custom container instead of the default scrollable body.
Doing so requires a ref to be passed which references the scrollable continer. However if the ref is passed down a component and
then the ref is furthur passed along then useScroll seems to fail.

https://www.framer.com/motion/use-scroll/

https://github.com/eeshankeni/famer-motion-scroll-bug

Steps to reproduce:

`yarn install`

`yarn dev` to run development mode. everything works as expected here.

`yarn build` to build static files.

`npx serve@latest out` to run the static page. Notice how the animation breaks here.

video demo:

https://github.com/framer/motion/assets/19875557/17c409a7-cc95-4bc3-bb79-e208339fb83a

**Comment by simplypixi** [comment]
Same issue here. Setting `layoutEffect` to `false` didn't fix the problem.
I've added extra logs and `scrollYProgress` does not change at all during a scroll. 

framer: 10.16
next: 13.5.6 (+css modules).

**Comment by Abdallah-Abdelkhalek** [comment]
const { scrollXProgress } = useScroll({
    layoutEffect: false, `<-- This line fixed the problem` 
    container: projectsContainerRef,
  });

**Comment by VictorAugDB** [comment]
> const { scrollXProgress } = useScroll({ layoutEffect: false, `<-- This line fixed the problem` container: projectsContainerRef, });

Yeah, this fixes the problem, but why does this problem only happen in prod?
This is because of SSR or SSG? I don't think so because I had the same problem and I was making use of the 'use client' directive in the entire page.
Is the page layout rendered differently in dev? Why? This should happen? Why there isn't anything in the framer motion docs warning about it?
I searched in the useScroll section in the docs and there isn't anything warning that this can happen, or detailing what this property does maybe I'm missing something but why the f* there isn't a search bar in the docs?

**Comment by CroAnna** [comment]
> const { scrollXProgress } = useScroll({ layoutEffect: false, `<-- This line fixed the problem` container: projectsContainerRef, });

this needs to be added in child component, if not clean from answer
